src/Symfony/Component/Notifier/Bridge/Mercure/MercureTransport.php

```diff
@@ -0,0 +1,101 @@
+use Symfony\Component\Notifier\Message\SentMessage;
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\Exception\HttpExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Mathias Arlaud <mathias.arlaud@gmail.com>
+ *
+ * @internal
+ *
+ * @experimental in 5.3
+ */
+final class MercureTransport extends AbstractTransport
+{
+    private $token;
+    private $hubPath;
+    private $topic;
+    private $secure;
+    private $scheme;
+
+    public function __construct(string $token, string $hubPath, ?string $topic = null, bool $secure = true, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null)
+    {
+        $this->token = $token;
+        $this->hubPath = $hubPath;
+        $this->topic = $topic;
+        $this->secure = $secure;
+        $this->scheme = $secure ? 'https' : 'http';
+
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('mercure://%s%s?topic=%s&secure=%s', $this->getEndpoint(), $this->hubPath, urlencode($this->topic), $this->secure ? 'true' : 'false');
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof ChatMessage && (null === $message->getOptions() || $message->getOptions() instanceof MercureOptions);
+    }
+
+    /**
+     * @see https://symfony.com/doc/current/mercure.html#publishing
+     */
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof ChatMessage) {
+            throw new LogicException(sprintf('The "%s" transport only supports instances of "%s" (instance of "%s" given).', __CLASS__, ChatMessage::class, get_debug_type($message)));
+        }
+
+        $topic = $message->getRecipientId() ?? $this->topic;
+        if (null === $topic) {
+            throw new RuntimeException(sprintf('Could not find any topic to publish the message. Please specify one using the "topic" DSN option or by passing a "%s" to the message.', MercureOptions::class));
+        }
+
+        // @see https://www.w3.org/TR/activitystreams-core/#jsonld
+        $update = new Update($message->getRecipientId() ?? $this->topic, json_encode([
+            '@context' => 'https://www.w3.org/ns/activitystreams',
+            'type' => 'Notification',
+            'summary' => $message->getSubject(),
+        ]));
+
+        $publisher = new Publisher(sprintf('%s://%s%s', $this->scheme, $this->getEndpoint(), $this->hubPath), new StaticJwtProvider($this->token), $this->client);
+
+        try {
+            $messageId = $publisher($update);
+        } catch (HttpExceptionInterface $e) {
+            throw new TransportException(sprintf('Unable to post the Mercure message: "%s".', $e->getResponse()->getContent(false)), $e->getResponse());
```

Comment:
You should pass the previous exception (and previous exception code) to ease debugging.